src/nvim/spellfile.c

```diff
@@ -5569,48 +5570,39 @@ static void init_spellfile(void)
       lstart = lend + 1;
     }
   }
+  
+  // If the spelllang points to a .spl file, use that path directly.
+  if (aspath) {
+    xmemcpyz(buf, curbuf->b_s.b_p_spl, (size_t)(lstart - curbuf->b_s.b_p_spl - 1));
+  } else {
+    buf = get_xdg_home(kXDGDataHome);
+  }
```

Comment:
would need to free `buf` before reassigning it here. But instead of doing that, change this to:

```suggestion
  char *buf = aspath? xmalloc(MAXPATHL) : get_xdg_home(kXDGDataHome);
```